src/datomish/exec.cljc

```diff
@@ -0,0 +1,67 @@
+(ns datomish.exec
+  #?(:cljs
+     (:require-macros
+      [datomish.pair-chan :refer [go-pair <?]]
+      [cljs.core.async.macros :refer [go]]))
+  (:require
+     [datomish.sqlite :as s]
+     [datomish.sqlite-schema :as ss]
+     [datomish.query :as dq]
+     #?@(:clj
+           [[datomish.jdbc-sqlite :as j]
+            [datomish.pair-chan :refer [go-pair <?]]
+            [clojure.core.async :refer [go <!! <! >! chan close! take!]]])
+     #?@(:cljs
+           [[datomish.promise-sqlite :as j]
+            [datomish.pair-chan]
+            [cljs.core.async :as a :refer [<!! <! >! chan close! take!]]])))
+
+(defn <run
+  "Execute the provided query on the provided DB.
+   Returns a transduced channel of results.
+   Closes the channel when fully consumed."
+  [db find]
+  (let [context (dq/find->prepared-context (dq/parse find))
+        row-transducer (dq/row-transducer context (dq/sql-projection context))
+        chan (chan 50 row-transducer)]
+
+    (s/<all-rows db (dq/context->sql-string context) chan)
+    chan))
+
+(defn channel->lazy-seq
+  "Returns a blocking lazy sequence of items taken from the provided channel."
+  [channel]
+  (lazy-seq
+   (when-let [v (<!! channel)]
+     (cons v (seq!! channel)))))
+
+(defn run-to-seq
+  "Given an open database, returns a lazy sequence of results.
+   When fully consumed, underlying resources will be released."
+  [db find]
+  (channel->lazy-seq (<run db find)))
+
+(defn <open-and-run-to-seq-promise
```

Comment:
Let's break some of these things into a `repl` module, since they're very unlikely to be the final API, but are super useful in hacking and testing.